pkg/roachprod/roachprod.go

```diff
@@ -66,6 +66,21 @@ const (
 	prometheusHostUrlEnv = "COCKROACH_PROM_HOST_URL"
 )
 
+// MalformedClusterNameError is returned when the cluster name passed to Create is invalid.
+type MalformedClusterNameError struct {
+	name        string
+	reason      string
+	suggestions []string
+}
+
+func (e *MalformedClusterNameError) Error() string {
+	return fmt.Sprintf("Malformed cluster name %s, %s. Did you mean one of %s", e.name, e.reason, e.suggestions)
+}
+
+// _findActiveAccounts is a test only variable, used for mocking a provider finding
+// active accounts in a unit test, where we don't want to actually access a provider.
+var _findActiveAccounts func(l *logger.Logger) (map[string]string, error)
```

Comment:
Is it okay to do this for a package variable? I couldn't think of an easier way to mock `vm.FindActiveAccounts`. AFAICT, the tests in a package are run sequentially, so no concurrency in accessing this.

Reply:
Yep, this is a totally legit. way of mocking which is also used sometimes for dependency injection. E.g., `InitTestConnectorFactory` [1] is used to inject (or mock) a test-specific implementation of the KV connector.

However, we could initialize it with the _actual_ implementation, i.e.,

```
var _findActiveAccounts = vm.FindActiveAccounts
```

and get rid of the conditional logic below.

[1] https://github.com/cockroachdb/cockroach/blob/6ec873aa0fc35a968e8daf3abd23ba895e4512fc/pkg/kv/kvclient/kvtenant/test_utils.go#L16

Reply:
Done! Thanks for tip